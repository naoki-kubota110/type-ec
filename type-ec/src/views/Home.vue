<template>
<v-app>
  <v-container>
    <v-main align="center">
      <Search/>
    </v-main>
    <v-row v-if="flg" class="mt-5">
      <v-col cpl="4" v-for="(item,index) in itemBox" :key="index" align="center">
        <v-card id="item" elevation="5">
          <router-link :to='{name: "ItemDetail", params: {id: item.id}}' id="router">
            <div>
              <v-img :src="require(`@/assets/img/${item.imagePath}`)" id="img" class="gray lighten-2"></v-img>
            </div>
            <div>
              <v-card-title>
                <p id="itemName">{{item.name}}</p>
              </v-card-title>
            </div>
          </router-link>
          <v-row>
            <v-spacer></v-spacer>
            <v-col cols="9">
              <div>
                <p>普通:{{item.priceM}}円  大盛り:{{item.priceL}}円</p>
              </div>
            </v-col>
          </v-row>
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</v-app>
</template>

<script lang="ts">
import { Component, Vue} from 'vue-property-decorator';
import Search from '../components/Search.vue'
@Component({
  components:{
    Search
  },
})
export default class Home extends Vue{
  created(){
    this.$store.state.flg = true
    this.$store.dispatch("fetchItem")
    this.$store.dispatch("fetchToppings")
  }
  get itemBox(){
    return this.$store.state.items;
  }
  get flg(){
    return this.$store.state.flg
  }
}
</script>

<style>
#img{
 height:200px;
 width:300px;
 border-bottom: solid 0.5px gray;
 border-radius:10px 10px 0 0;
 background-color: white;
}
#item{
  height:325px;
  width:300px;
  border-radius: 10px;
  border: solid 0.5px gray;
  background-color: antiquewhite;
}
#price{
  font-size: 25px;
}
#itemName{
  color:black;
  text-decoration: none;
  font-size: 18px;
  font-weight:bold;
}
#itemName:hover{
  color:orange;
}
#router{
   text-decoration: none;
}
</style>