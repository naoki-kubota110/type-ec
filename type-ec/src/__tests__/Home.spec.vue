<script lang="ts">
import Vue from 'vue'
import Vuetify from 'vuetify'
import Home from 'views/Home.vue';
import VueRouter from 'vue-router'
import { createLocalVue, mount,RouterLinkStub } from '@vue/test-utils'
import assert from 'assert'
Vue.use(Vuetify)
const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()


describe('Counter', () => {
  const localVue = createLocalVue()
  let vuetify
   beforeEach(() => {
    vuetify = new Vuetify()
  })
  it('v-cartタグの数が１つかの確認', () => {
    const wrapper = mount(Home,{
      localVue,
      vuetify
    });
    expect(wrapper.findAll('.v-card').length).toEqual(1)
  });
  it('card-contentが3つ存在しているのかの確認', () => {
    const wrapper = mount(Home,{
      localVue,
      vuetify
    });
    expect(wrapper.findAll('.homeMain').length).toEqual(1)
  });
  it('v-showがtrueにっているのかの確認', () => {
    const wrapper = mount(Home,{
      localVue,
      vuetify
    });
    expect(wrapper.isVisible()).toBeTruthy();
  });
});



</script>