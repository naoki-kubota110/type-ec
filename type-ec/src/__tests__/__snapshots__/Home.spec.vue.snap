// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter is a Vue instance 1`] = `
"<div data-app=\\"true\\" class=\\"v-application v-application--is-ltr theme--light\\" id=\\"app\\">
  <div class=\\"v-application--wrap\\">
    <div class=\\"container\\">
      <main class=\\"v-main\\" style=\\"padding: 0px 0px 0px 0px;\\" align=\\"center\\">
        <div class=\\"v-main__wrap\\">
          <div class=\\"container\\">
            <div class=\\"row justify-center\\">
              <div class=\\"col-sm-10 col-md-8 col-lg-6 col-12\\">
                <div class=\\"v-card v-sheet theme--light\\">
                  <div id=\\"cardTitle\\" class=\\"v-card__title\\">商品を検索する</div>
                  <hr role=\\"separator\\" aria-orientation=\\"horizontal\\" class=\\"v-divider theme--light\\">
                  <div class=\\"v-card__text\\">
                    <form novalidate=\\"novalidate\\" class=\\"v-form\\">
                      <div align=\\"center\\">
                        <div class=\\"v-input v-input--hide-details theme--light v-text-field\\">
                          <div class=\\"v-input__control\\">
                            <div class=\\"v-input__slot\\">
                              <div class=\\"v-text-field__slot\\"><label for=\\"input-8\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">商品名 or キーワードで検索</label><input id=\\"input-8\\" type=\\"text\\"></div>
                            </div>
                          </div>
                        </div>
                        <p></p>
                        <!----> <button type=\\"button\\" class=\\"v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default\\" style=\\"background-color: rgb(61, 57, 53); border-color: #3d3935;\\"><span class=\\"v-btn__content\\">検索</span></button>&nbsp;
                        <button type=\\"button\\" class=\\"v-btn v-btn--outlined theme--light v-size--default\\"><span class=\\"v-btn__content\\">クリア</span></button>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
            <!---->
          </div>
        </div>
      </main>
      <!---->
    </div>
  </div>
</div>"
`;
